public/index.js

```diff
@@ -175,6 +175,7 @@
     var emailAddress = emailRecipient.value;
     var http = new XMLHttpRequest();
     http.open("POST", '/finished', true);
-    http.send(emailAddress);
+    var payload = JSON.stringify([emailAddress, sessionStorage]);
```

Comment:
Sorry to have missed this, but you need to make set a header (`application/json`) on this request.
Also, this should not be an array. It should be an object: 
```js
  {  emailAddress: emailAddress, emailData: sessionStorage }
```